### PR TITLE
task #2787:- multiple device id support

### PR DESCRIPTION
**Pull request by shreyas1434shinde**

### CODE REVIEW

File: src/components/Header.tsx [hunk 1]
```diff
@@ -70,13 +72,42 @@ const Header: React.FC<HeaderProps> = ({ toggleDrawer, openDrawer }) => {
...
-  const handleLogoutClick = () => {
+  const handleLogoutClick = async() => {
     router.replace('/logout');
     logEvent({
       action: 'logout-clicked-header',
       category: 'Dashboard',
       label: 'Logout Clicked',
     });
+    const token = localStorage.getItem('token');
+    const userResponse = await getUserId();
```
Comment (itsvick) [inline src/components/Header.tsx:83]: @shreyas1434shinde are u calling api for getting user id?

File: src/components/Header.tsx [hunk 2]
```diff
@@ -70,13 +72,42 @@ const Header: React.FC<HeaderProps> = ({ toggleDrawer, openDrawer }) => {
...
+    // Retrieve deviceID from local storage
+    const deviceID = localStorage.getItem('deviceID');
+    if (deviceID) {
+      try {
+        // Update device notification
+        const headers = {
+          tenantId: userResponse?.tenantData[0]?.tenantId,
```
Comment (itsvick) [inline src/components/Header.tsx:90]: @shreyas1434shinde just use the axiosInstance with post method instead of manually adding headers here

**Review by itsvick — CHANGES_REQUESTED** [review]

File: src/components/Header.tsx (hunk 2 above)
Comment (shreyas1434shinde) [reply src/components/Header.tsx]: resolved

File: src/components/PlannedSession.tsx [hunk 3]
```diff
@@ -993,7 +993,10 @@ const PlannedSession: React.FC<PlannedModalProps> = ({
                     });
 
                     if (response?.result?.userDetails) {
-                      const deviceId = response?.result?.userDetails.map((device: any) => device?.deviceId).filter((id: any) => id !== null);
+                      const deviceId = response?.result?.userDetails
```
Comment (itsvick) [inline src/components/PlannedSession.tsx:996]: @shreyas1434shinde can u check this `const deviceId = response?.result?.userDetails
  ?.flatMap((device: { deviceId?: string | null }) => device.deviceId || [])
  .filter(Boolean);`

File: src/components/SessionCard.tsx [hunk 4]
```diff
@@ -76,7 +76,7 @@ const SessionsCard: React.FC<SessionsCardProps> = ({
         const deviceIds = response.result.userDetails
           .filter((user: any) => user.role === Role.TEACHER || user.role === Role.STUDENT)
           .map((user: any) => user.deviceId)
-          .filter((id: any) => id !== null);
+          .filter((id: any) => id !== null).flat();;
```
Comment (itsvick) [inline src/components/SessionCard.tsx:79]: `const deviceIds = response.result.userDetails
  .filter((user: { role: Role; deviceId?: string | null }) =>
    [Role.TEACHER, Role.STUDENT].includes(user.role)
  )
  .flatMap((user) => user.deviceId || []);`

@shreyas1434shinde

**Pull request merged**